core/src/main/java/hudson/model/Slave.java

```diff
@@ -133,10 +133,7 @@
     private /*almost final*/ DescribableList<NodeProperty<?>,NodePropertyDescriptor> nodeProperties = 
                                     new DescribableList<NodeProperty<?>,NodePropertyDescriptor>(Jenkins.getInstance().getNodesObject());
 
-    /**
-     * Lazily computed set of labels from {@link #label}.
-     */
-    private transient volatile Set<Label> labels;
```

Comment:
I fear you can't touch it because it may cause issues as i definitely remember that using labels cache for slaves. Feel free to investigate cloud plugins.